HOTT-4629: Improve exception handling

rasikasri opened the pull request:
### Jira link

[HOTT-4629](https://transformuk.atlassian.net/browse/HOTT-4629)

### What?

I have added/removed/altered:

- Added actions in the controller for the missing error-handling actions
- added generic view for errors: `error.html.erb` view

### Why?

I am doing this because:

- There are certain exceptions in which are mapped in the default Rails behaviour to actions we don’t currently support

Commit:
HOTT-4629: Extend Frontend ErrorsController

Commit:
HOTT-4629: Extend Frontend ErrorsController

Commit:
Merge remote-tracking branch 'origin/HOTT-4629-error-controller' into HOTT-4629-error-controller

# Conflicts:
#	app/controllers/errors_controller.rb

alexdesi committed:
HOTT-4629: Add logic to handle more exceptions.

willfish commented on app/controllers/errors_controller.rb:
```diff
@@ -15,6 +15,17 @@ def not_found
     end
   end
 
+  # issue: this one is not handled by the controller 
```
Remove me

willfish commented on app/models/exchange_rates/exchange_rate.rb:
```diff
@@ -17,7 +17,7 @@ def formatted_validity_start_date
   end
 
-  def formatted_validity_end_date
+  def formatted_validity_end_datee
```
```suggestion
  def formatted_validity_end_date
```

alexdesi committed:
HOTT-4629: Add spec application_controller_spec.
Removed redundant views for error messages (reusing error.html.erb)
Support more errors in the routes.

alexdesi commented on config/environments/development.rb:
```diff
@@ -11,8 +11,11 @@
+
   # Show full error reports.
-  config.consider_all_requests_local = true
+  config.consider_all_requests_local = false
```
The application will render the public error pages instead of detailed debug information.

alexdesi replied:
It has pros and cons:
It makes it easier to manually test the behaviour as production.
On the other side, for certain debugging, we should set it to false.

alexdesi commented on config/environments/test.rb:
```diff
@@ -22,14 +22,14 @@
+
   # Show full error reports and disable caching.
-  config.consider_all_requests_local       = true
+  config.consider_all_requests_local       = false
```
The application will render the public error pages instead of detailed debug information.

alexdesi commented on spec/requests/application_controller_spec.rb:
```diff
@@ -0,0 +1,51 @@
+RSpec.describe ApplicationController, type: :request do
```
I reconsidered the usefulness of these specs.
We can remove these. We are just (re)testing the mappings in the rails actionpack/lib/action_dispatch/middleware/exception_wrapper.rb are correct.

The useful tests are the `spec/requests/errors_controller_spec.rb`,
since the tests the routes and the rendered views.

alexdesi committed:
HOTT-5629: Refactor skip_news_banner.
Set consider_all_requests_local to true on Dev (to see detailed error reports).

alexdesi committed:
HOTT-4629: Fix pages_controller_spec tests.

alexdesi committed:
HOTT-4629: Fix glossary controller specs.

willfish committed:
HOTT-4629: Code review

willfish commented on config/environments/development.rb:
```diff
@@ -11,6 +11,9 @@
   config.eager_load = false
 
+  # Raise exceptions instead of rendering exception templates.
+  config.action_dispatch.show_exceptions = true
```
I think we can remove this chipolata?

willfish replied:
Nope, I'm wrong
